crates/pathfinder/src/state/sync.rs

```diff
@@ -123,65 +126,19 @@ where
+
+    // Delay before restarting L1 or L2 tasks if they fail.
+    static START_DELAY: std::time::Duration = std::time::Duration::from_secs(60);
 
     loop {
         tokio::select! {
             l1_event = rx_l1.recv() => match l1_event {
-                Some(l1::Event::Update(updates)) => {
-                    let first = updates.first().map(|u| u.block_number.get());
-                    let last = updates.last().map(|u| u.block_number.get());
-
-                    l1_update(&mut db_conn, &updates).await.with_context(|| {
-                        format!("Update L1 state with blocks {first:?}-{last:?}")
-                    })?;
-
-                    match updates.as_slice() {
-                        [single] => {
-                            tracing::info!("L1 sync updated to block {}", single.block_number);
-                        }
-                        [first, .., last] => {
-                            tracing::info!(
-                                "L1 sync updated with blocks {} - {}",
-                                first.block_number,
-                                last.block_number
-                            );
-                        }
-                        _ => {}
-                    }
-                }
-                Some(l1::Event::Reorg(reorg_tail)) => {
-                    l1_reorg(&mut db_conn, reorg_tail)
-                        .await
-                        .with_context(|| format!("Reorg L1 state to block {reorg_tail}"))?;
-
-                    let new_head = match reorg_tail {
-                        StarknetBlockNumber::GENESIS => None,
-                        other => Some(other - 1),
-                    };
-
-                    match new_head {
-                        Some(head) => {
-                            tracing::info!("L1 reorg occurred, new L1 head is block {}", head)
-                        }
-                        None => tracing::info!("L1 reorg occurred, new L1 head is genesis"),
-                    }
-                }
-                Some(l1::Event::QueryUpdate(block, tx)) => {
-                    let update =
-                        tokio::task::block_in_place(|| {
-                            let tx = db_conn.transaction()?;
-                            L1StateTable::get(&tx, block.into())
-                        })
-                        .with_context(|| format!("Query L1 state table for block {block:?}"))?;
-
-                    let _ = tx.send(update);
-
-                    tracing::trace!("Query for L1 update for block {}", block);
-                }
+                Some(update) => {
+                    let tx = db_conn.transaction().context("Create database transaction")?;
+                    L1StateTable::upsert(&tx, &update).context("Upsert l1_state")?;
+                    tx.commit().context("Commit database transaction")?;
+                    tracing::info!("L1 state update: block={} root={}", update.block_number, update.global_root);
```

Comment:
```suggestion
                    tracing::info!(block=%update.block_number, state_commitment=update.global_root, "L1 state update: block={} root={}", update.block_number, update.global_root);
```

Reply:
Am I doing something wrong? When I try to apply the suggestion, I get 
`error[E0277]: the trait bound `primitive_types::H256: tracing::Value` is not satisfied`

Reply:
No I'm just bad at typing :) edited the suggestion again.

Its also possible that `H256` does not implement `Display` -- either way, I think you will probably want to change the `L1StateUpdate.global_root` to be a `StateCommitment` not `H256` -- or at least do that change somewhere. I missed that in my first review.